src/js/core/methods/CardanoSignTransaction.js

```diff
@@ -114,7 +125,35 @@ export default class CardanoSignTransaction extends AbstractMethod {
         };
     }
 
+    _ensureFeatureIsSupported(feature: $Keys<typeof CardanoSignTransactionFeatures>) {
+        if (!this.device.atLeast(CardanoSignTransactionFeatures[feature])) {
+            throw ERRORS.TypedError('Method_InvalidParameter', `Feature ${feature} not supported by device firmware`)
+        }
+    }
+
+    _ensureFirmwareSupportsParams() {
+        const params = this.params;
+
+        params.certificates.map((certificate) => {
+            if (certificate.type === CERTIFICATE_TYPE.StakePoolRegistration) {
+                this._ensureFeatureIsSupported('SignStakePoolRegistrationAsOwner');
+            }
+        });
+
+        if (params.validity_interval_start != null) {
```

Comment:
```suggestion
        if (typeof params.validity_interval_start === 'number') {
```
!= null does not exclude `undefined` case

Reply:
not sure if I got you, the purpose of the condition is to trigger when `params.validity_interval_start` is defined, be it null or undefined.

```
const x = {}
console.log(x.foo != null)
```
prints false which is desired. `undefined ` would be left out if the operator was `!==`

Reply:
ah, now i see it `validityIntervalStart` is a string (number represented as string, right?)
maybe it should be validated properly in line 39 (as 'amount' instead of 'string')?

otherwise im able to set param validityIntervalStart: 'foo-bar'

Reply:
Initially I set it to string as it didn't feel right to mix it with true amounts but validation-wise they are the same, so I agree. Maybe an alias for `uint64` for generic unsigned big numbers in paramsValidator would be nice, but it's not a big deal after all